### DISCUSSION TITLE
Design decision behind grid

### DISCUSSION THREAD

**Discussion by jspaezp** [wfondrie/mokapot#88] [General]
Hello there!

I was working on an extension and noticed the definition of the grid search for the percolator model ... and I wondered 2 things.

Is there any difference between elements of the grid that have proportional coefficients ...
In other words, would `0:0.1, 1:1` be different from `0:1,1:10`? Since both assign 10x more value to the classification/classification of the elements in `1`.

This is because removing those repeated ratios would reduce the number of model training iterations from 9 to 5!

LMK what you think!
-Sebastian

Code for future reference:

https://github.com/wfondrie/mokapot/blob/46fc0328f52055d5c73c83c56da76c3e7078cf5b/mokapot/model.py#L33-L37

```
>>> PERC_GRID = {
...     "class_weight": [
...         {0: neg, 1: pos} for neg in (0.1, 1, 10) for pos in (0.1, 1, 10)
...     ]
... }
>>> 
>>> PERC_GRID
{'class_weight': [{0: 0.1, 1: 0.1}, {0: 0.1, 1: 1}, {0: 0.1, 1: 10}, {0: 1, 1: 0.1}, {0: 1, 1: 1}, {0: 1, 1: 10}, {0: 10, 1: 0.1}, {0: 10, 1: 1}, {0: 10, 1: 10}]}
>>> sorted([x[0]/x[1] for x in PERC_GRID['class_weight']])
[0.01, 0.1, 0.1, 1.0, 1.0, 1.0, 10.0, 10.0, 100.0]
>>> len([x[0]/x[1] for x in PERC_GRID['class_weight']])
9
>>> len(set([x[0]/x[1] for x in PERC_GRID['class_weight']]))
5
>>> PERC_GRID = {
...     "class_weight": [
...         {0: neg, 1: pos} for neg in (0.1, 1, 10) for pos in (0.1, 1, 10) if not (pos == neg and not pos == 1)
...     ]
... }
>>> PERC_GRID
{'class_weight': [{0: 0.1, 1: 1}, {0: 0.1, 1: 10}, {0: 1, 1: 0.1}, {0: 1, 1: 1}, {0: 1, 1: 10}, {0: 10, 1: 0.1}, {0: 10, 1: 1}]}
```

**Comment by wfondrie** [comment]
> Is there any difference between elements of the grid that have proportional coefficients ...
In other words, would 0:0.1, 1:1 be different from 0:1,1:10?

Surprising, yes there is in this case. For the SVC model, the `class_weights` parameter controls the weight of the regularization parameter, `C`, that controls the margin in the hinge loss function. `{0: 0.1, 1: 1}` results in a relatively low tolerance for incorrect classifications from both classes, whereas `{0: 1, 1: 10}` penalizes less for misclassified examples in either class by comparison.
